src/simple_cache_client.rs

```diff
@@ -1873,6 +1873,132 @@ impl SimpleCacheClient {
+        &mut self,
+        cache_name: &str,
+        set_name: impl IntoBytes,
+        order: sorted_set::Order,
+        range: impl RangeBounds<i32>,
+        with_scores: bool,
+    ) -> MomentoResult<SortedSetFetch> {
+        // transforms the Rust `Range` start into a Momento start index. Because
+        // the Momento start index is always Inclusive (or Unbounded) we need to
+        // map the value of a Rust `Excluded` start bound to an
+        // `InclusiveStartIndex` by adding one to the value.
+        let start = match range.start_bound() {
+            std::ops::Bound::Included(v) => {
+                sorted_set_fetch_request::by_index::Start::InclusiveStartIndex(*v)
+            }
+            std::ops::Bound::Excluded(v) => {
+                sorted_set_fetch_request::by_index::Start::InclusiveStartIndex(*v + 1)
+            }
+            std::ops::Bound::Unbounded => {
+                sorted_set_fetch_request::by_index::Start::UnboundedStart(Unbounded {})
+            }
+        };
+
+        // transforms the Rust `Range` end into a Momento end index. Because the
+        // Momento end index is always Exclusive (or Unbounded) we need to map
+        // the value of a Rust `Included` end bound to an `ExclusiveEndIndex` by
+        // adding one to the value.
+        let end = match range.end_bound() {
+            std::ops::Bound::Included(v) => {
+                sorted_set_fetch_request::by_index::End::ExclusiveEndIndex(*v + 1)
+            }
+            std::ops::Bound::Excluded(v) => {
+                sorted_set_fetch_request::by_index::End::ExclusiveEndIndex(*v)
+            }
+            std::ops::Bound::Unbounded => {
+                sorted_set_fetch_request::by_index::End::UnboundedEnd(Unbounded {})
+            }
+        };
+
+        let request = self.prep_request(
+            cache_name,
+            SortedSetFetchRequest {
+                set_name: set_name.into_bytes(),
+                order: order.into(),
+                with_scores,
+                range: Some(sorted_set_fetch_request::Range::ByIndex(
+                    sorted_set_fetch_request::ByIndex {
+                        start: Some(start),
+                        end: Some(end),
+                    },
+                )),
+            },
+        )?;
+
+        let response = self
+            .data_client
+            .sorted_set_fetch(request)
+            .await?
+            .into_inner();
+
+        // this flattens the response returning a None for both a missing sorted
+        // set and for a request that returns Found with elements being None and
+        // converting the SortedSet enum into the interior collection of
+        // elements.
+        match response.sorted_set {
+            Some(crate::sorted_set::SortedSet::Found(elements)) => match elements.elements {
+                Some(elements) => Ok(SortedSetFetch::Found { elements }),
+                None => Ok(SortedSetFetch::Missing),
```

Comment:
I agree with you on the flattening but I think we would just return a `Hit` with an empty vec in this case.